cmd/importer/main.go

```diff
@@ -0,0 +1,183 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/cmd/importer/pod"
+	"sigs.k8s.io/kueue/cmd/importer/util"
+	"sigs.k8s.io/kueue/pkg/util/useragent"
+)
+
+const (
+	NamespaceFlag      = "namespace"
+	NamespaceFlagShort = "n"
+	QueueMappingFlag   = "queuemapping"
+	QueueLabelFlag     = "queuelabel"
+	QPSFlag            = "qps"
+	BurstFlag          = "burst"
+	VerbosityFlag      = "verbose"
+	VerboseFlagShort   = "v"
+	JobsFlag           = "jobs"
+	JobsFlagShort      = "j"
+)
+
+var (
+	k8sClient client.Client
+	rootCmd   = &cobra.Command{
+		Use:   "importer",
+		Short: "Import existing (running) objects into Kueue",
+		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
+			v, _ := cmd.Flags().GetCount(VerbosityFlag)
+			level := (v + 1) * -1
+			ctrl.SetLogger(zap.New(
+				zap.UseDevMode(true),
+				zap.ConsoleEncoder(),
+				zap.Level(zapcore.Level(level)),
+			))
+
+			kubeConfig, err := ctrl.GetConfig()
+			if err != nil {
+				return err
+			}
+			if kubeConfig.UserAgent == "" {
+				kubeConfig.UserAgent = useragent.Default()
+			}
+			qps, err := cmd.Flags().GetFloat32(QPSFlag)
+			if err != nil {
+				return err
+			}
+			kubeConfig.QPS = qps
+			bust, err := cmd.Flags().GetInt(BurstFlag)
+			if err != nil {
+				return err
+			}
+			kubeConfig.Burst = bust
+
+			if err := kueue.AddToScheme(scheme.Scheme); err != nil {
+				return err
+			}
+
+			c, err := client.New(kubeConfig, client.Options{Scheme: scheme.Scheme})
+			if err != nil {
+				return err
+			}
+			k8sClient = c
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Flags().Visit(func(f *pflag.Flag) {
+				fmt.Println(f.Name, f.Value.String())
+			})
+			err := checkCmd(cmd, args)
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringSliceP(NamespaceFlag, NamespaceFlagShort, nil, "target namespaces (at least one should be provided)")
+	rootCmd.MarkPersistentFlagRequired(NamespaceFlag)
+
+	rootCmd.PersistentFlags().String(QueueLabelFlag, "", "label used to identify the target local queue")
+	rootCmd.PersistentFlags().StringToString(QueueMappingFlag, nil, "mapping from \""+QueueLabelFlag+"\" label values to local queue names")
```

Comment:
This will be very hard to use or edit.

Instead, accept a filepath that has the mappings in a simple yaml format.

Or it could be a single file for at least "queuelabel" and "queuemapping". Then the user can provide it via a ConfigMap.

Reply:
Also missing the flavor name.

Reply:
Added mapping from file

> Also missing the flavor name.

The flavor used in the admission is the first flavor of the the CQ, the user should define that